crates/transaction-pool/src/blobstore/mem.rs

```diff
@@ -113,6 +114,31 @@ impl BlobStore for InMemoryBlobStore {
         Ok(items)
     }
 
+    fn get_by_versioned_hashes(
+        &self,
+        versioned_hashes: &[B256],
+    ) -> Result<Vec<Option<BlobAndProofV1>>, BlobStoreError> {
+        let mut result = vec![None; versioned_hashes.len()];
+        for (_tx_hash, blob_sidecar) in self.inner.store.read().iter() {
```

Comment:
Whilst correct, linearly iterating the pool might be a less-than-ideal solution long run?